lib/standards.h

```diff
@@ -35,10 +35,10 @@
  */
 struct Standards {
     /** C code standard */
-    enum cstd_t { C89, C99, C11, CLatest = C11 } c = CLatest;
+    enum cstd_t { C89, C99, C11, CLatest = 99 } c = CLatest;
```

Comment:
why do you assign a value here `CLatest=99` ?

Reply:
So we have a fixed value for the "latest" standard which does not change when we introduce a new one.

Currently the latest is C++20. So if you store "latest" as the desired platform instead of an explicit one it won't be the latest anymore if you we add C++23 and will still point to C++20 since both are represented by the same value.

Reply:
Where is the raw value for CLatest used?

Spontanously I am guessing that maybe the GUI Settings store the raw integer value? If the GUI settings store the raw integer value of the enum maybe that is a mistake and it should save the corresponding string instead.

I believe it breaks a misra rule to only assign last constant in such enum declaration. I intend to add a CI check that cppcheck source code is compliant to Misra C++ and Cert C++, using Cppcheck premium, with exceptions for rules that we do not like. Well we could assign all the constants in this enum fixed values that would be Misra compliant but that feels unfortunate to me.

Reply:
> Where is the raw value for CLatest used?
> 
> Spontanously I am guessing that maybe the GUI Settings store the raw integer value? If the GUI settings store the raw integer value of the enum maybe that is a mistake and it should save the corresponding string instead.

That was guess as well but I think we are actually storing a string (i.e. `c++20`). So if `CppLatest` is used we would store the specific string instead. So that needs an adjustment to actually store `latest` and handle as a string.

That would also be similar to the MSVC command-line option `/std:latest`.